src/rmkit/fb/fb.cpy

```diff
@@ -256,19 +265,30 @@ namespace framebuffer:
       if o_y >= self.height || o_x >= self.width || o_y < 0 || o_x < 0:
         return
 
-      update_dirty(dirty_area, o_x, o_y)
-      update_dirty(dirty_area, o_x+w, o_y+h)
+      if likely(fill):
+        for j 0 h:
+          if j+o_y >= self.height:
+            break
 
-      for j 0 h:
-        if j+o_y >= self.height:
-          break
+          for i 0 w:
+            if i+o_x >= self.width:
+              break
```

Comment:
I wonder how expensive these comparisons are (they might not be at all) compared to:

```cpp
w = min(w, self.width-x)
h = min(h, self.height-x)
for j 0 h:
  for i 0 w:
    do_dithering(self.fbmem, i+o_x, j+o_y, color, dither)
```

Reply:
i had a little trouble with this just now (and had trouble the first couple times i tried it a while back)